internal/grpc_dialer_test.go

```diff
@@ -199,31 +202,57 @@ func TestCustomResolver(t *testing.T) {
 	require.NoError(t, err)
 	defer client.Close()
 
+	// Round-robin appears to apply to transport _connections_ rather than just
+	// addresses. As such we spin here until we have round-tripped an RPC to
+	// both servers to guarantee that connections to both have been established.
+	// This test can fail spuriously without this section as the calls to
+	// SignalWorkflow below will race with grpc-go's connection establishment.
+	// This technique is consistent with the approach used in the grpc-go
+	// codebase itself:
+	// https://github.com/grpc/grpc-go/blob/bd7076973b45b81e37a45eb761efb789e2001618/balancer/roundrobin/roundrobin_test.go#L196-L212
+	connected := map[net.Addr]struct{}{}
+	req := workflowservice.SignalWorkflowExecutionRequest{
+		WorkflowExecution: &common.WorkflowExecution{WorkflowId: "workflowid", RunId: "runid"},
+		SignalName:        "signal",
+		Namespace:         DefaultNamespace,
+		Identity:          t.Name(),
+	}
+	var peerOut peer.Peer
+	for len(connected) < 2 {
```

Comment:
`connected` is always empty here, right?

Reply:
It starts as empty but the body of the loop inserts into it

Reply:
Sorry, I misread! Thanks!